### PR TITLE
【KernelGen】refactor(sparse_mla): vLLM-compatible 3D interface with dual-kernel dispatch

### PR DESCRIPTION
**Pull request by Schopenhauer-loves-Hegel**
## Summary

Rewrite `fused/DSA/sparse_mla.py` with vLLM-compatible 3D interface and dual-kernel dispatch for optimal decode + prefill performance.

### Interface Change

Replaced FlagGems 4D interface `(B, SQ, H, DT)` with vLLM 3D interface:
- `q: [s_q, h_q, d_qk]` — Query tensor
- `kv: [s_kv, h_kv, d_qk]` — KV cache
- `indices: [s_q, h_kv, topk]` — Sparse attention indices
- Returns `(output, max_logits, lse)` matching `vllm._flashmla_C.sparse_prefill_fwd`

### Optimizations

- **Dual-kernel dispatch**: Decode kernel (`s_q <= 128`) with explicit `out_dtype` for lower latency; Prefill kernel (`s_q > 128`) with eviction hints for better throughput
- **Late scaling**: Scale QK after dot products (avoids bf16→fp32→bf16 conversion)
- **FP32 accumulation**: Improved numerical stability
- **Removed hardcoded `is_causal`**: Sparse attention with explicit indices does not need causal masking
- **28 autotune configs**: 18 for prefill + 10 for decode, covering BK={64,128}, BH={16,32,64}, various warp/stage combos

### Performance

26 test cases (13 configs × decode + prefill), single GPU pinned via `CUDA_VISIBLE_DEVICES`, results verified across multiple runs.

#### H20

| Config | Decode (s_q=1) | Prefill (s_q=4096) |
|--------|---------------|-------------------|
| GLM5-C1 (h_q=64, topk=512) | ~0.96x | 0.65x |
| GLM5-C2 (h_q=128, topk=1024) | **1.49x** | 0.66x |
| DeepSeek-V3.2 (h_q=128, topk=2048) | **1.61x** | 0.64x |
| **Overall avg** | | **1.01x** |

#### H100 80GB

| Config | Decode (s_q=1) | Prefill (s_q=4096) |
|--------|---------------|-------------------|
| GLM5-C1 (h_q=64, topk=512) | 0.72x | 0.52x |
| GLM5-C2 (h_q=128, topk=1024) | 0.81x | 0.52x |
| DeepSeek-V3.2 (h_q=128, topk=2048) | 0.80x | 0.50x |
| **Overall avg** | | **0.65x** |

**Key findings:**
- On H20, decode (GLM5-C2, DSV3) **outperforms vLLM CUDA kernel** by 1.49-1.61x
- On H20, overall average reaches **1.01x** (on par with hand-written CUDA)
- On H100, overall average is **0.65x** (CUDA kernel has larger advantage on stronger hardware)
- All 26 accuracy tests pass (max_diff < 0.01 vs PyTorch reference)

### Test Changes

- Replaced FlagGems 4D test format with vLLM 3D format
- Real model configs: GLM-5 CONFIG1, GLM-5 CONFIG2, DeepSeek-V3.2
- Both decode (s_q=1) and prefill (s_q=4096) test cases
- Added benchmark tests (`test_sparse_mla_benchmark_decode`, `test_sparse_mla_benchmark_prefill`)
- Pure PyTorch chunked reference implementation for correctness verification

## Test Plan

- [x] 26 accuracy cases pass (13 configs × 2 s_q values)
- [x] Edge case tests pass
- [x] No NaN in any output
- [x] Decode + prefill benchmark tests added
- [x] Verified on H20 and H100 with consistent results across multiple runs
- [ ] Run FlagGems CI tests

### CODE REVIEW

File: src/flag_gems/fused/DSA/sparse_mla.py [hunk 1]
```diff
@@ -1,218 +1,415 @@
...
+    Uses v99 for s_q <= SQ_THRESHOLD (decode), v91 otherwise (prefill).
+
+    Args:
+        q:       [s_q, h_q, d_qk]       bf16
+        kv:      [s_kv, h_kv, d_qk]     bf16
+        indices: [s_q, h_kv, topk]       int32
+    Returns:
+        (output, max_logits, lse)
+    """
+    s_q, h_q, d_qk = q.shape
```
Comment (tengqm) [inline src/flag_gems/fused/DSA/sparse_mla.py:385]: should we log the operator name here?

File: tests/test_DSA/test_sparse_mla_ops.py [hunk 2]
```diff
@@ -1,272 +1,136 @@
...
-    ), f"Output shape incorrect: {your_output.shape} != {expected_shape}"
+    assert not torch.isnan(act_out).any(), "Output contains NaN"
+    max_diff = (ref_out.float() - act_out.float()).abs().max().item()
+    assert max_diff < 1e-2, f"Output max diff {max_diff:.6f} exceeds tolerance 1e-2"
 
 
 if __name__ == "__main__":
-    # Can directly run this file for testing
-    pytest.main([__file__, "-v"])
+    pytest.main([__file__, "-v", "-s"])
```
Comment (tengqm) [inline tests/test_DSA/test_sparse_mla_ops.py]: We usually don't include a separate main in test cases.

File: src/flag_gems/fused/DSA/sparse_mla.py (hunk 1 above)
Comment (lukatao) [reply src/flag_gems/fused/DSA/sparse_mla.py:385]: updated, add log GEMS SPARSE MLA

File: tests/test_DSA/test_sparse_mla_ops.py (hunk 2 above)
Comment (lukatao) [reply tests/test_DSA/test_sparse_mla_ops.py]: removed

**Comment by bin913** [comment]
@Schopenhauer-loves-Hegel  Please fix failing check and resolve conflicts

**Comment by bin913** [comment]
@Schopenhauer-loves-Hegel May I ask if you are still working on this PR?
